woocommerce/woocommerce-ios
# Product Creation AI: Generate product name from keywords

Closes: #10737


## Description

This PR updates the Yosemite layer to add a new action for generating product names based on keywords and integrate it to the UI. I introduced a new feature type for AI: `woo_ios_product_name` to use for the feature.

I also noticed the incorrect feature used when identifying language for product description generation. Updated it since it's a small change.

## Testing instructions

- Create a JN site with Jetpack AI plugin. (Checkbox available in JN home page to install this plugin)
- Switch to Products tab, select the "+" button to add a new product. The same AI action sheet should be displayed.
- Tap on "Create a product with AI" to launch "Add your product name" screen
- Tap on "Suggest a name", another sheet for generating product name should be displayed.
- Enter some keywords in the text field and tap Generate. After a bit, a generated name should be returned.
- Tap Regenerate to generate another name if needed.
- Tap Copy, the content should be copied to the pasteboard and a notice is displayed saying "Copied!".
- Tap Apply, the generated name should be filled in the add product name screen.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/4a290a3a-7669-4405-87e1-784adabb9317



---
